### PR TITLE
Fix circular import for `PageInterface` when using custom page interface

### PR DESCRIPTION
**Pull request by mgax**
I'd like to define a custom `PageInterface` (using the [`PAGE_INTERFACE `](https://wagtail-grapple.readthedocs.io/en/latest/getting-started/settings.html#page-interface) setting). Because `grapple.types.pages.PageInterface` is quite complex, I'd like to inherit from it, and add some additional fields.

This results in a circular import, because the definition of the `Page` class in the same module calls `get_page_interface`: https://github.com/torchbox/wagtail-grapple/blob/d9cd8f272836fbf914485d6af4b6c19c9b1589b1/grapple/types/pages.py#L165-L173

---
This PR moves `PageInterface` and `get_page_interface` to a new module, `interfaces.py`.

### CODE REVIEW

**Comment by mgax** [comment]
CI is failing for the _coverage_ job, but it looks like all recent CI runs are failing on that.

**Comment by zerolab** [comment]
the `actions/upload-artifact` step needs `include-hidden-files: true` ([example](https://github.com/wagtail/Willow/blob/main/.github/workflows/test.yml#L59), [reference](https://github.blog/changelog/2024-08-19-notice-of-upcoming-deprecations-and-breaking-changes-in-github-actions-runners/))

File: tests/test_interfaces.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 from testapp.factories import BlogPageFactory, CustomInterfaceBlockFactory
 from testapp.interfaces import CustomInterface
 
-from grapple.types.pages import PageInterface, get_page_interface
+from grapple.types.interfaces import PageInterface, get_page_interface
```
Comment (zerolab) [inline tests/test_interfaces.py:10]: can you update `testapp.interfaces.CustomInterface` to inherit from `PageInterface`?

we have `test_get_page_interface_with_custom_page_interface` but it was not fully testing your scenario

**Comment by zerolab** [comment]
Since we're making these changes, it may be worth moving the `StreamFieldInterface` in the new interfaces module

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -56,6 +56,7 @@ jobs:
           path: tests/.coverage*
           if-no-files-found: ignore
           retention-days: 1
+          include-hidden-files: true
```
Comment (zerolab) [inline .github/workflows/ci.yml:59]: there's another one below (line 100something)

File: tests/test_interfaces.py (hunk 1 above)
Comment (mgax) [reply tests/test_interfaces.py:10]: I've created a new interface, `CustomPageInterface`, for this purpose; `CustomInterface` is applied to types that are not pages and so things break.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mgax) [reply .github/workflows/ci.yml:59]: Ah, missed that, thanks!

**Comment by mgax** [comment]
> Since we're making these changes, it may be worth moving the `StreamFieldInterface` in the new interfaces module

Ah, I meant to do that, but got carried away. Thanks!

File: grapple/utils.py [hunk 3]
```diff
@@ -248,3 +248,38 @@ def get_media_item_url(cls):
...
+    rtn_obj = {}
+
+    if hasattr(obj, "raw_data"):
+        rtn_obj = []
+        for field in obj[0]:
+            rtn_obj.append(serialize_struct_obj(field.value))
+    # This conditionnal and below support both Wagtail >= 2.13 and <2.12 versions.
+    # The "stream_data" check can be dropped once 2.11 is not supported anymore.
+    # Cf: https://docs.wagtail.io/en/stable/releases/2.12.html#stream-data-on-streamfield-values-is-deprecated
+    elif hasattr(obj, "stream_data"):
```
Comment (zerolab) [inline grapple/utils.py:263]: note to self: follow up with clean-up PR

**Review by zerolab — APPROVED** [review]
LGTM

**Pull request merged**